modelskill/comparison/_comparison.py

```diff
@@ -0,0 +1,1223 @@
+        * y            (y) float64 50.6 51.66 52.7 53.75 54.8
+        Data variables:
+            n            (x, y) int32 3 0 0 14 37 17 50 36 72 ... 0 0 15 20 0 0 0 28 76
+            bias         (x, y) float64 -0.02626 nan nan ... nan 0.06785 -0.1143
+
+        >>> ds = cc.spatial_skill(binsize=0.5)
+        >>> ds.coords
+        Coordinates:
+            observation   'alti'
+        * x            (x) float64 -1.5 -0.5 0.5 1.5 2.5 3.5 4.5 5.5 6.5 7.5
+        * y            (y) float64 51.5 52.5 53.5 54.5 55.5 56.5
+        """
+
+        # TODO remove in v1.1
+        model, start, end, area = _get_deprecated_args(kwargs)
+
+        cmp = self.sel(
+            model=model,
+            start=start,
+            end=end,
+            area=area,
+        )
+
+        metrics = _parse_metric(metrics, self.metrics, return_list=True)
+        if cmp.n_points == 0:
+            raise ValueError("No data to compare")
+
+        df = cmp.to_dataframe()
+        df = _add_spatial_grid_to_df(df=df, bins=bins, binsize=binsize)
+
+        # n_models = len(df.model.unique())
+        # n_obs = len(df.observation.unique())
+
+        # n_obs=1 because we only have one observation (**SingleObsComparer**)
+        by = _parse_groupby(by=by, n_models=cmp.n_models, n_obs=1)
+        if isinstance(by, str) or (not isinstance(by, Iterable)):
+            by = [by]
+        if "x" not in by:
+            by.insert(0, "x")
+        if "y" not in by:
+            by.insert(0, "y")
+
+        df = df.drop(columns=["x", "y"]).rename(columns=dict(xBin="x", yBin="y"))
+        res = _groupby_df(df, by, metrics, n_min)
+        return SpatialSkill(res.to_xarray().squeeze())
+
+    @property
+    def residual(self):
+        return self.mod - np.vstack(self.obs)
+
+    def remove_bias(self, correct="Model"):
+        bias = self.residual.mean(axis=0)
+        if correct == "Model":
+            for j in range(self.n_models):
+                mod_name = self.mod_names[j]
+                mod_df = self.raw_mod_data[mod_name]
+                mod_df[mod_name] = mod_df.values - bias[j]
+                self.data[mod_name] = self.data[mod_name] - bias[j]
+        elif correct == "Observation":
+            # what if multiple models?
+            self.data[self._obs_name] = self.obs + bias
+        else:
+            raise ValueError(
+                f"Unknown correct={correct}. Only know 'Model' and 'Observation'"
+            )
+        return bias
+
+    # TODO remove plotting methods in v1.1
+    def scatter(
+        self,
+        *,
+        bins=20,
+        quantiles=None,
+        fit_to_quantiles=False,
+        show_points=None,
+        show_hist=None,
+        show_density=None,
+        norm=None,
+        backend="matplotlib",
+        figsize=(8, 8),
+        xlim=None,
+        ylim=None,
+        reg_method="ols",
+        title=None,
+        xlabel=None,
+        ylabel=None,
+        binsize=None,
```

Comment:
These three arguments are not used. Delete?

Reply:
The three I refer to are:

        binsize=None,
        nbins=None,
        skill_table=None,

Reply:
I just now see that the same goes for hist() below - maybe **kwargs it all?